README.md

```diff
@@ -792,52 +891,242 @@ This adds the following additional output when triggering `runCodec ()`:-
 
     Codec "<Snipped>" Could not decode "Exported" in "Favorites-ClientB" {event="{ \"count\": 2 }"}
 
-## Decoding contextual information
+<a name="metadata"></a>
+## Adding metadata to events based on Domain-intrinsic information
+
+The following recipe can be used to populate the `Meta` field of an event based on information your application supplies within Events it generates:
+
+```fsharp
+module StoreWithMeta =
+
+    type Event<'E> = int64 * Metadata * 'E
+    and Metadata = { principal: string }
+    and Codec<'E when 'E :> TypeShape.UnionContract.IUnionContract> = FsCodec.IEventCodec<Event<'E>, Store.EventBody, unit>
+
+    // we assume no special requirements for complex data types when deserializing the metadata, so use Default Options
+    let private options = Options.Default
+    // no special requirements for deserializing metadata
+    let private serdes = Serdes Options.Default
+    let codec<'E when 'E :> TypeShape.UnionContract.IUnionContract> : Codec<'E> =
+        // here we surface the metadata from the raw event as part of the application level event based on the stored form
+        let up struct (raw : Store.Event, contract : 'E) : Event<'E> =
+            raw.Index, serdes.Deserialize<Metadata> raw.Meta, contract
+        // _index: up and down are expected to encode/decode symmetrically - when encoding, the app supplies a dummy, and the store assigns it on appending
+        // the metadata is encoded as the normal bodies are
+        let down ((_index, meta : Metadata, event : 'E) : Event<'E>) =
+            struct (event, ValueSome meta, ValueNone)
+        CodecJsonElement.Create<Event<'E>, 'E, Metadata>(up, down, options = options) 
+```
+
+The above embeds and/or extracts contextual information from the Event's `Meta` field.
+
+NOTE this works well as long as the information in question is generated naturally as part of the application's processing,
+and it is relevant in the context of all operations within a Service.
+Where this is not the case (e.g., if you are attempting to add out of band contextual causation/correlation information that
+is external to the application's logic, see [Context](#context].
+
+<a name="context"></a>
+## Adding Metadata to Events based on extrinsic Context
+
```

Comment:
This is a good section. I'd only dug into how to write metadata and that I needed to pass that context when I resolve the decider.. It would be useful to link to this from the Equinox documentation.

Reply:
Thanks. Yes, this strongly inspired by production code that the authors had to suffer through writing without the aid of this documentation. Good point about the link from Equinox being important.

Reply:
Unfortunately I'm not seeing an obvious place to put it in the Equinox docs. Would appreciate if you could suggest somewhere (esp if you have a wording that would have spoken to you) in the context of https://github.com/jet/equinox/pull/380